### ISSUE TITLE
Multiple code chunks in SYNOPSIS should not interfere with each other

### ISSUE THREAD

**Issue by zoffixznet** [miyagawa/test-synopsis#12]
If you got something like:

````
Description

       SYNOPSIS_CODE_1

More description

       SYNOPSIS_CODE_2
```

Then code in `SYNOPSIS_CODE_1` block interferes with code in `SYNOPSIS_CODE_2` block. This should not be happening as the blocks are likely showing separate code examples.

**Comment by zoffixznet** [comment]
Here's a test POD that must pass when the issue is fixed:
```pod
=pod

=head1 SYNOPSIS

=for test_synopsis use warnings FATAL => 'all';

First sample code:

    my $x = 42;

Second sample code:

    my $x = 43;

=head1 DESCRIPTION

bleh

=cut
```

**Comment by zoffixznet** [comment]
Closed by d26775f
